paasta_tools/native_mesos_scheduler.py

```diff
@@ -174,11 +181,22 @@ def start_task(self, driver, offer):
 
         while self.need_more_tasks(base_task.name) and \
                 remainingCpus >= task_cpus and \
-                remainingMem >= task_mem:
+                remainingMem >= task_mem and \
+                len(remainingPorts) >= 1:
+
+            task_port = random.choice(list(remainingPorts))
```

Comment:
Marathon switched to random ports in mesosphere/marathon#1552. This makes it less likely that a container will come up on the same port as a recently-killed container for a different service and get traffic destined for the incorrect service. (This could happen if smartstack is delayed.)